test/blackbox-tests/test-cases/pkg/source-stanza/unknown-dune-version.t

```diff
@@ -0,0 +1,35 @@
+We are unable to pin projects that the version of dune doesn't understand.
+
+  $ . ../helpers.sh
+  $ mkrepo
+  $ add_mock_repo_if_needed
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.13)
+  > (package_source
+  >  (url "$PWD/_b")
+  >  (package (name a)))
+  > (package
+  >  (name main)
+  >  (depends a))
+  > EOF
+
+  $ mkdir _b
+  $ cat >_b/dune-project <<EOF
+  > (lang dune 100.1)
+  > ;; one day we'll get here
+  > EOF
+
+# The location here is messed up b/c we are using a source path (incorrectly)
+# to construct the project
+
+  $ dune pkg lock
+  File "dune-project", line 1, characters 11-16:
+  1 | (lang dune 3.13)
```

Comment:
Yes, unfortunately this is quite difficult to fix. We can only create a dune-project with `Path.Source.t` but this can't be used for correct locations.